app/models/service.rb

```diff
@@ -97,11 +97,11 @@ def direct_vms
   end
 
   def all_vms
-    MiqPreloader.preload_and_map(subtree, :direct_vms)
+    MiqPreloader.preload_and_map(subtree, :vms)
```

Comment:
@kbrock so it does a lot of queries that are cached in master. But it does N+1 queries at least for OrchestrationStack and OrchestrationTemplate and their VMs